File: TC_pruebas_login/test_playwright/tc_04.py
from playwright.sync_api import sync_playwright
from var import *

counter = 1

def tomar_screenshot(page, screen_name: str):
    global counter
    page.screenshot(path=f"TC_pruebas_login/test_playwright/test_images/tc_04_{screen_name}_{counter}.png")
    counter += 1

with sync_playwright() as p:
    browser = p.chromium.launch(headless=False)
    page = browser.new_page()

    page.goto(link)
    page.wait_for_timeout(3000)
    tomar_screenshot(page, "logout_exitoso")

    page.locator(uname_input).fill("tomsmith")
    page.locator(pw_input).fill("SuperSecretPassword!")
    tomar_screenshot(page, "credenciales")

    page.locator(btn_continue).click()
    page.wait_for_timeout(2000)
    page.locator('//*[@id="content"]/div/h4')
    tomar_screenshot(page, "login_exitoso")

    try:
        btn_logout = page.locator('//*[@id="content"]/div/a/i')
        btn_logout.click()
        page.wait_for_timeout(2000)

        logout_text = page.locator('//*[@id="flash"]')
        text = logout_text.text_content()
        tomar_screenshot(page, "logout_exitoso")

        if "You logged out of the secure area!" in text:
            print(f"Test exitoso: {text}")
        else:
            print("Test fallido: Mensaje de éxito no encontrado")
    except Exception as e:
        print(f"Test fallido: Ocurrió un error - {e}")

    browser.close()

File: TC_pruebas_login/test_puppetter/tc_04.js
const puppeteer = require('puppeteer');

// Variables
const link = "https://the-internet.herokuapp.com/login";
const uname_input = '#username';
const pw_input = '#password';
const btn_continue = '#login button i';
const btn_logout = '#content a i';
const flash_message = '#flash';
const secure_area = '#content h4';

let counter = 1;

async function tomar_screenshot(page, screen_name) {
    await page.screenshot({
        path: `test_images/tc_04_${screen_name}_${counter}.png`
    });
    counter++;
}

async function sleep(ms) {
    return new Promise(resolve => setTimeout(resolve, ms));
}

(async () => {
    const browser = await puppeteer.launch({ headless: false });
    const page = await browser.newPage();

    await page.goto(link, { waitUntil: 'networkidle2' });
    await sleep(3000);
    await tomar_screenshot(page, "inicio");

    // Login con credenciales válidas
    await page.type(uname_input, "tomsmith");
    await page.type(pw_input, "SuperSecretPassword!");
    await tomar_screenshot(page, "credenciales");

    await page.click(btn_continue);
    await sleep(2000);

    // Verificar login exitoso
    await page.$(secure_area);
    await tomar_screenshot(page, "login_exitoso");

    try {
        // Logout
        await page.click(btn_logout);
        await sleep(2000);

        const logout_message = await page.$(flash_message);
        if (logout_message) {
            const text = await page.evaluate(el => el.textContent, logout_message);
            await tomar_screenshot(page, "logout_exitoso");

            if (text.includes("You logged out of the secure area!")) {
                console.log(`✅ Test exitoso: ${text}`);
            } else {
                console.log("❌ Test fallido: Mensaje de éxito no encontrado");
            }
        } else {
            console.log("❌ Test fallido: No se encontró el mensaje de logout");
        }
    } catch (e) {
        console.log(`❌ Test fallido: Ocurrió un error - ${e}`);
    }

    await browser.close();
})();
